test/tf_exec.cpp

```diff
@@ -360,15 +379,110 @@ TEST(tf_exec, Tile) {
-  ASSERT_EQ(outputs_C[0].shape(),outputs_C_cpu[0].shape());
-  ASSERT_EQ(outputs_D[0].shape(),outputs_D_cpu[0].shape());
-  AssertTensorEquals(outputs_C[0],outputs_C_cpu[0]);
-  AssertTensorEquals(outputs_D[0],outputs_D_cpu[0]);
+  ASSERT_EQ(outputs_C[0].shape(), outputs_C_cpu[0].shape());
+  ASSERT_EQ(outputs_D[0].shape(), outputs_D_cpu[0].shape());
+  AssertTensorEquals(outputs_C[0], outputs_C_cpu[0]);
+  AssertTensorEquals(outputs_D[0], outputs_D_cpu[0]);
 }
 
+TEST(tf_exec, Op_Conv2DBackpropFilter) {
+  tf::Scope root = tf::Scope::NewRootScope();
+  tf::Scope root_ngraph = root.NewSubScope("sub_scope_ngraph");
+  root_ngraph = root_ngraph.WithDevice("/device:NGRAPH:0");
+
+  // TF Default formats
+  // Input NHWC :[batch, in_height, in_width, in_channels]
+  std::vector<tf::int64> input_size_NHWC = {1, 7, 6, 2};
+  // Filter :[filter_height, filter_width, in_channels, out_channels]
+  std::vector<tf::int64> filter_size_HWIO = {3, 3, 2, 2};
+  // Out_delta :[batch, out_height, out_width, out_channels]
+  std::vector<tf::int64> output_del_size_valid = {1, 3, 2, 2};
+  std::vector<tf::int64> output_del_size_same = {1, 4, 3, 2};
+  tf::Tensor output_delta_valid(tf::DT_FLOAT,
+                                tf::TensorShape(output_del_size_valid));
+  tf::Tensor output_delta_same(tf::DT_FLOAT,
+                               tf::TensorShape(output_del_size_same));
+  AssignInputValues(output_delta_valid);
+  AssignInputValues(output_delta_same);
+
+  std::map<std::string, tf::Tensor*> out_delta_size_map = {
+      {"VALID", &output_delta_valid}, {"SAME", &output_delta_same}};
+
+  std::vector<int> stride = {1, 2, 2, 1};
+  tf::Tensor input_data(tf::DT_FLOAT, tf::TensorShape(input_size_NHWC));
+  AssignInputValues(input_data);
+
+  auto filter_sizes = tf::ops::Const(root, {3, 3, 2, 2});
+
+  tf::ClientSession session(root);
+  std::vector<tf::Tensor> outputs_ngraph;
+  std::vector<tf::Tensor> outputs_cpu;
+
+  // TEST NHWC : default data format
+  for (auto map_iterator : out_delta_size_map) {
+    auto padding_type = map_iterator.first;
+    auto output_delta = *(out_delta_size_map[padding_type]);
+
+    auto r_ngraph = tf::ops::Conv2DBackpropFilter(
+        root_ngraph.WithOpName("r_NGRAPH"), input_data, filter_sizes,
+        output_delta, stride, padding_type);
+
+    auto r_cpu = tf::ops::Conv2DBackpropFilter(
+        root.WithOpName("r_CPU"), input_data, filter_sizes, output_delta,
+        stride, padding_type);
+
+    TF_CHECK_OK(session.Run({r_ngraph}, &outputs_ngraph));
+    TF_CHECK_OK(session.Run({r_cpu}, &outputs_cpu));
+
+    ASSERT_EQ(outputs_ngraph[0].shape(), outputs_cpu[0].shape());
+    AssertTensorEquals(outputs_ngraph[0], outputs_cpu[0]);
+  }
+
+  // TEST NCHW
+  // Dialtion rates > 1 not supported on CPU
+  // Current testing only with dialtion rate 1
+  tf::ops::Conv2DBackpropFilter::Attrs op_attr_nchw;
+  op_attr_nchw = op_attr_nchw.DataFormat("NCHW");
+  op_attr_nchw = op_attr_nchw.Dilations({1, 1, 1, 1});
```

Comment:
Why is `Dilations` attrs only set for `NCHW` case not in `NHWC` case?

Reply:
the tf::ops::Conv2DBackpropFilter::Attrs() is an optional argument. If not passed, the default attrs sets data_format NHWC and dilations [1,1,1,1]. So the NHWC test actually runs with the dilation[1,1,1,1] but it is not explicitly passed.